File: RL/Part-I/src/Constraint.py
class Constraint:
    def __init__(self, Events, rew, index, name):
        self.Events = Events
        self.reward = rew
        self.index = index
        self.name = name

    def __repr__(self):
        return "Cons: ( " + str(self.index) + " " + str(self.Events) + " " + str(self.reward) + " " + str(self.name) + " )"

    def __eq__(self, other):
        if isinstance(other, self.__class__):
            return self.index == other.index and self.reward == other.reward and self.name==other.name
        return False

File: EM and NLP/src/main/java/Constraint.java
import java.util.ArrayList;

/**
 * Created by tarun on 5/6/17.
 */
public class Constraint {
    ArrayList<Event> events;
    double reward;
    int index;

    public Constraint(ArrayList<Event> events, double rew, int ind) {
        this.events = events;
        this.reward = rew;
        this.index = ind;
    }

    @Override
    public boolean equals(Object o) {
        if (!(o instanceof Constraint)) {
            return false;
        }
        Constraint other = (Constraint) o;

        return this.index==other.index &&
                this.reward==other.reward;
    }
}
